images/miq-app/ose-assets/ose-deploy-common.sh

```diff
@@ -0,0 +1,217 @@
+#!/bin/bash
+
+[[ -s /etc/default/evm ]] && source /etc/default/evm
+
+# This file is created by the write_deployment_info during initial deployment
+PV_DEPLOY_INFO_FILE="${APP_ROOT_PERSISTENT}/.deployment_info"
+
+# This directory is used to store initialization logfiles on PV
+PV_LOG_DIR="${APP_ROOT_PERSISTENT}/log"
+
+# This file is supplied by the app docker image with default files/dirs to persist on PV
+CONTAINER_DATA_PERSIST_FILE="/container.data.persist"
+
+# Copy of CONTAINER_DATA_PERSIST_FILE that will be stored on PV and can be customized by users to add more files/dirs
+PV_DATA_PERSIST_FILE="$APP_ROOT_PERSISTENT/container.data.persist"
+
+# VMDB app_root directory inside persistent volume mount
+APP_ROOT_PERSISTENT_VMDB=${APP_ROOT_PERSISTENT}/var/www/miq/vmdb
+
+function check_deployment_status() {
+# Description
+# Inspect PV for previous deployments, if a DB a config is present, restore
+# Source previous deployment info file from PV and compare data with current environment
+# Evaluate conditions and decide a target deployment type: redeploy,upgrade or new
+
+echo "== Checking deployment status =="
+
+if [[ -f ${APP_ROOT_PERSISTENT_VMDB}/config/database.yml && -f ${PV_DEPLOY_INFO_FILE} ]]; then
+  echo "== Found existing deployment configuration =="
+  echo "== Restoring existing database configuration =="
+  ln --backup -sn ${APP_ROOT_PERSISTENT_VMDB}/config/database.yml ${APP_ROOT}/config/database.yml
+  # Source original deployment info variables from PV
+  source ${PV_DEPLOY_INFO_FILE}
+  # Obtain current running environment
+  APP_VERSION=$(cat ${APP_ROOT}/VERSION)
+  SCHEMA_VERSION=$(cd ${APP_ROOT} && RAILS_USE_MEMORY_STORE=true bin/rake db:version | awk '{ print $3 }')
+  # Check if we have identical EVM versions (exclude master builds)
+  if [[ $APP_VERSION == $PV_APP_VERSION && $APP_VERSION != master ]]; then
+    echo "== App version matches original deployment =="
+  # Check if we have same schema version for same EVM version
+    if [ "${SCHEMA_VERSION}" != "${PV_SCHEMA_VERSION}" ]; then
+       echo "ERROR: Something seems wrong, db schema version mismatch for the same app version: ${PV_SCHEMA_VERSION} <-> ${SCHEMA_VERSION}"
+       exit 1
+    fi
+  # Assuming redeployment (same APP_VERSION)
+  export DEPLOYMENT_STATUS=redeployment
+  else
+  # Assuming upgrade (different APP_VERSION)
+  export DEPLOYMENT_STATUS=upgrade
```

Comment:
So far is an upgrade when APP_VERSION != PV_APP_VERSION, we need run extended checks for UPGRADE to ensure we are not upgrading to an older APP_VERSION but given the versioning situation between upstream/downsteam, need to discuss how to handle non-numeric vs numeric...